src/datasets/load.py

```diff
@@ -452,8 +452,12 @@ def prepare_packaged_module(name):
 
     if force_local_path is None:
         main_folder_path = os.path.join(datasets_modules_path if dataset else metrics_modules_path, short_name)
+        # Create the lock file where we know we have write permissions.
+        lock_path = main_folder_path + ".lock"
+        os.makedirs(main_folder_path, exist_ok=True)
```

Comment:
Ah thank you!

Something like:

```python
# Create the lock file where we know we have write permissions.
lock_path = (datasets_modules_path if dataset else metrics_modules_path) + f"{name}.lock"
```

so that each dataset has their own lock file?